mysk-lib/src/models/club/db.rs

```diff
@@ -1,36 +1,50 @@
 use crate::{
     common::{
-        requests::{
-            FilterConfig, PaginationConfig, QueryablePlaceholder, SortablePlaceholder,
-            SortingConfig,
-        },
+        requests::{FilterConfig, PaginationConfig, QueryParam, SortingConfig, SqlSection},
         response::PaginationType,
     },
     error::Error,
     helpers::date::get_current_academic_year,
     models::{
+        club::request::{queryable::QueryableClub, sortable::SortableClub},
         enums::{ActivityDayHouse, SubmissionStatus},
-        traits::QueryDb,
+        traits::{QueryDb, Queryable as _},
     },
     prelude::*,
 };
 use chrono::{DateTime, Utc};
 use mysk_lib_derives::{BaseQuery, GetById};
 use mysk_lib_macros::traits::db::{BaseQuery, GetById};
 use serde::Deserialize;
-use sqlx::{query, FromRow, PgPool, Postgres, QueryBuilder};
+use sqlx::{query, FromRow, PgPool, Postgres, QueryBuilder, Row as _};
 use uuid::Uuid;
 
 #[derive(BaseQuery, Clone, Debug, Deserialize, FromRow, GetById)]
-#[base_query(query = "SELECT * FROM clubs")]
+#[base_query(
+    query = "
```

Comment:
I suggest creating a view for clubs with this query maybe name it as `detailed_club_view`

Reply:
At the end of the day that view's query is going to use an `INNER JOIN` anyways, so why don't we use the abstractions here instead?

Reply:
Abstraction on the database level is easier to change and keep track of than on the codebase level

Reply:
Okay, well here are my counter arguments:
1. Both the teacher and student data models are still using joins in their base queries.
2. Compared to `elective_subject_sessions_with_detailed_view`'s query, club's query is not as complicated that which I personally feel that makes it good enough to be included in the codebase instead.

But, I guess it's your call at the end of the day.

Reply:
Well, the club's supposed to include member_count from compact onwards but that's not required for porting the original codebase, personally all the inherited style tables are supposed to be made a views so data can be query in the same way both using the supabase's SQL editor and using the codebase